service/dataSourceService.go

```diff
@@ -269,32 +271,65 @@ func (service *DataSourceService) GetHarvestStatus(ctx context.Context, id strin
 func (service *DataSourceService) ConsumeReport(ctx context.Context, routingKey string, body []byte) []error {
 	var errors []error
 	if strings.Contains(routingKey, "harvested") {
-		var reports []model.HarvestReport
-		err := json.Unmarshal(body, &reports)
-		if err != nil {
-			errors = append(errors, err)
-		} else {
-			for _, report := range reports {
-				err := service.ReportsRepository.UpsertReports(ctx, report)
-				if err != nil {
-					errors = append(errors, err)
-				}
+		errors = service.consumeHarvestedReport(ctx, body)
+	} else if strings.Contains(routingKey, "reasoned") {
+		errors = service.consumeReasonedReport(ctx, body)
+	} else if strings.Contains(routingKey, "ingested") {
+		errors = service.consumeIngestedReport(ctx, routingKey, body)
+	}
+
+	return errors
+}
+
+func (service *DataSourceService) consumeHarvestedReport(ctx context.Context, body []byte) []error {
+	var errors []error
+	var reports []model.HarvestReport
+	err := json.Unmarshal(body, &reports)
+	if err != nil {
+		errors = append(errors, err)
+	} else {
+		for _, report := range reports {
+			err := service.ReportsRepository.UpsertReports(ctx, report)
+			if err != nil {
+				errors = append(errors, err)
 			}
 		}
-	} else if strings.Contains(routingKey, "reasoned") || strings.Contains(routingKey, "ingested") {
-		var startAndEnd model.StartAndEndTime
-		err := json.Unmarshal(body, &startAndEnd)
+		}
+	return errors
+}
+
+func (service *DataSourceService) consumeReasonedReport(ctx context.Context, body []byte) []error {
+	var errors []error
+	var reports []model.HarvestReport
+	err := json.Unmarshal(body, &reports)
+	if err != nil {
+		errors = append(errors, err)
+	} else {
+		for _, report := range reports {
+			report.ID = ReasoningReportID(report.ID)
+			err := service.ReportsRepository.UpsertReports(ctx, report)
+			if err != nil {
+				errors = append(errors, err)
+			}
+		}
+	}
+	return errors
```

Comment:
Dette skiller unmarshal feil fra upsert feil. Vanlig i go å returnere feil med en gang, fremfor if-else og return til slutt.

```suggestion
	var reports []model.HarvestReport
	err := json.Unmarshal(body, &reports)
	if err != nil {
		return []error{err}
	}
	
	var errors []error
	for _, report := range reports {
		report.ID = ReasoningReportID(report.ID)
		err := service.ReportsRepository.UpsertReports(ctx, report)
		if err != nil {
			errors = append(errors, err)
		}
	}
	return errors
```

Reply:
886d945